File: Blind150/2.Two Pointer/Python/odd_even_parity_2.py
# https://leetcode.com/problems/sort-array-by-parity-ii/description/

from typing import List
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:


        e, o = 0 , 1


        while e < len(nums) - 1:


            while nums[e] % 2 != 0:

                nums[e], nums[o] = nums[o], nums[e]

                o += 2

            e += 2

        return nums

File: Blind150/2.Two Pointer/GO/odd_even_parity_2.go
func sortArrayByParityII(nums []int) []int {

    e,o := 0,  1

    for e < len(nums)-1{

        for nums[e] % 2 != 0  && o < len(nums) {

            nums[e], nums[o] = nums[o], nums[e]
            o = o + 2

        }

        e = e + 2


    }

    return nums

}
